File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& nums, int k) {
        k=nums.size()-k;
        int right=0;
        int left=0;
        int sum=0;
        int max_score=0;
        int total_points=accumulate(nums.begin(),nums.end(),0);
        while(right<nums.size()){
            sum+=nums[right];
            while(left<right && (right-left+1)>k){
                sum-=nums[left];
                left++;
            }
            if((right-left+1)==k){
                max_score=max(max_score,total_points-sum);
            }
            right++;

        }
        return max_score==0?total_points:max_score;

        
    }
};

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int n=cardPoints.length;
        int size=n-k;
        int left=0;
        int right=0;
        int totalSum=0;
        for(int ele:cardPoints)totalSum+=ele;
        int sum=0;
        int maxi=Integer.MIN_VALUE;
        while(right<n){
            sum+=cardPoints[right];
            while((right-left+1)>(size)){
                sum-=cardPoints[left];
                left++;
            }
            if((right-left+1)==size)
            {
                maxi=Math.max(maxi,totalSum-sum);
            }
            right++;
        }
        return maxi;
        
    }
}